### PR TITLE
bane component to element. nullrod now uses it for revenants.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Title. A MOB_LIVING_BANED signal has been added. Revenants use it for extra effects beyond damage after being baned by nullrods.

## Why It's Good For The Game
It's a fairly simple component, has no ref variable and no proc that messes up with the values of our variable.
todo in a later PR: remove `check_species_weakness()` and use the bane element in their place.

## Changelog
N/A

### CODE REVIEW

**Comment by KittyNoodle** [comment]
Hey could you merge these into one pr, five prs doing the same thing seems kinda excessive, just change them to commits in one large pr

**Comment by Ghommie** [comment]
> 
> 
> Hey could you merge these into one pr, five prs doing the same thing seems kinda excessive, just change them to commits in one large pr

They are different components that do different things. I'm just atomizing the different changes to facilitate the merging process.

**Comment by Ryll-Ryll** [comment]
bane?

File: code/datums/elements/bane.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/// Deals extra damage to mobs of a certain type or species.
+/datum/element/bane
+	element_flags = ELEMENT_DETACH|ELEMENT_BESPOKE
+	id_arg_index = 2
+	/// can be a mob or a species.
+	var/target_type
+	/// multiplier of the item strength for damage calculation.
+	var/damage_multiplier
```
Comment (ATH1909) [inline code/datums/elements/bane.dm:8]: might want to mention that this is only for the EXTRA damage, so you need to use a value of 1 here if you want a bane item to do 2x damage, a value of 2 if you want it to do 3x damage, etc.

File: code/modules/antagonists/revenant/revenant.dm [hunk 2]
```diff
@@ -191,15 +192,12 @@
...
-/mob/living/simple_animal/revenant/attackby(obj/item/W, mob/living/user, params)
-	. = ..()
-	if(istype(W, /obj/item/nullrod))
-		visible_message(span_warning("[src] violently flinches!"), \
-						span_revendanger("As \the [W] passes through you, you feel your essence draining away!"))
-		adjustBruteLoss(25) //hella effective
-		inhibited = TRUE
-		update_action_buttons_icon()
-		addtimer(CALLBACK(src, .proc/reset_inhibit), 30)
+/mob/living/simple_animal/revenant/proc/on_baned(obj/item/weapon, mob/living/user)
```
Comment (Mothblocks) [inline code/modules/antagonists/revenant/revenant.dm:195]: SIGNAL_HANDLER

File: code/modules/antagonists/revenant/revenant.dm [hunk 3]
```diff
@@ -191,15 +192,12 @@
...
+	visible_message(span_warning("[src] violently flinches!"), \
+					span_revendanger("As \the [weapon] passes through you, you feel your essence draining away!"))
```
Comment (Mothblocks) [inline code/modules/antagonists/revenant/revenant.dm:197]:
```suggestion
		span_revendanger("As [weapon] passes through you, you feel your essence draining away!"))
```

File: code/modules/antagonists/revenant/revenant.dm [hunk 4]
```diff
@@ -191,15 +192,12 @@
...
+	inhibited = TRUE
+	update_action_buttons_icon()
+	addtimer(CALLBACK(src, .proc/reset_inhibit), 30)
```
Comment (Mothblocks) [inline code/modules/antagonists/revenant/revenant.dm:200]:
```suggestion
	addtimer(CALLBACK(src, .proc/reset_inhibit), 3 SECONDS)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
This doesn't really look cleaner as an element, but it's not that much worse so eh.

**Pull request merged**